src/main/java/com/exasol/containers/status/ContainerStatus.java

```diff
@@ -0,0 +1,111 @@
+package com.exasol.containers.status;
+
+import java.io.Serializable;
+import java.util.EnumMap;
+import java.util.Map;
+
+import javax.annotation.processing.Generated;
+
+import com.exasol.containers.ExasolService;
+
+/**
+ * State information about the Exasol docker container.
+ */
+public final class ContainerStatus implements Serializable {
+    private static final long serialVersionUID = 5014025686797381100L;
+    private final String containerId;
+    private final Map<ExasolService, ServiceStatus> serviceStatuses = new EnumMap<>(ExasolService.class);
+
+    private ContainerStatus(final String containerId) {
+        this.containerId = containerId;
+        for (final ExasolService service : ExasolService.values()) {
+            this.serviceStatuses.put(service, ServiceStatus.NOT_CHECKED);
+        }
+    }
+
+    /**
+     * Create container state for the container with the given ID.
+     *
+     * @param containerId docker container ID
+     * @return container state
+     */
+    public static ContainerStatus create(final String containerId) {
+        return new ContainerStatus(containerId);
+    }
+
+    /**
+     * Get the ID of the container this status belongs to.
+     *
+     * @return container ID
+     */
+    public String getContainerId() {
+        return this.containerId;
+    }
+
+    /**
+     * Get the status of the given service.
+     *
+     * @param service service for which the status should be provided
+     * @return status of the service
+     */
+    public ServiceStatus getServiceStatus(final ExasolService service) {
+        return this.serviceStatuses.get(service);
+    }
+
+    /**
+     * Check if a service is ready.
+     *
+     * @param service service to be checked
+     * @return {@code true} if the service is ready
+     */
+    public boolean isServiceReady(final ExasolService service) {
+        return this.serviceStatuses.get(service) == ServiceStatus.READY;
+    }
+
+    /**
+     * Set the status for a given service
+     *
+     * @param service       service for which the status should be changed
+     * @param serviceStatus new status
+     */
+    public void setServiceStatus(final ExasolService service, final ServiceStatus serviceStatus) {
+        this.serviceStatuses.put(service, serviceStatus);
+    }
+
+    @Generated("org.eclipse.Eclipse")
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = (prime * result) + ((this.containerId == null) ? 0 : this.containerId.hashCode());
```

Comment:
why not `Objects.hash( containerId, ...)`
that's much more readable

Reply:
Auto-generated, so I did not much care. Switched the generator settings now.